spec/rubocop/cop/lint/require_relative_self_path_spec.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Lint::RequireRelativeSelfPath, :config do
+  it 'registers an offense when using `require_relative` with self file path argument' do
+    expect_offense(<<~RUBY, 'foo.rb')
+      require_relative 'foo'
+      ^^^^^^^^^^^^^^^^^^^^^^ Remove the `require_relative` that requires itself.
+      require_relative 'bar'
+    RUBY
+
+    expect_correction(<<~RUBY)
+      require_relative 'bar'
+    RUBY
+  end
+
+  it 'registers an offense when using `require_relative` with self file path argument (with ext)' do
+    expect_offense(<<~RUBY, 'foo.rb')
+      require_relative 'foo.rb'
+      ^^^^^^^^^^^^^^^^^^^^^^^^^ Remove the `require_relative` that requires itself.
+      require_relative 'bar'
+    RUBY
+
+    expect_correction(<<~RUBY)
+      require_relative 'bar'
+    RUBY
+  end
+
+  it 'does not register an offense when using `require_relative` without self file path argument' do
+    expect_no_offenses(<<~RUBY, 'foo.rb')
+      require_relative 'bar'
+    RUBY
+  end
+
+  it 'does not register an offense when using `require_relative` without argument' do
+    expect_no_offenses(<<~RUBY, 'foo.rb')
+      require_relative
```

Comment:
This will always raise `ArgumentError` and hence should be detected by rubocop. Maybe we change the cop name to `Lint/RequireRelativePath` and accomodate this case too?

Reply:
This test case has confirmed that it does not cause an error with a valid grammar for Ruby. And I think it should do method argument checking with RBS instead of RuboCop, so the cop name and role will remain the same.